src/components/incident/test/IncidentDetails.test.tsx

```diff
@@ -0,0 +1,181 @@
+        const durationValue = formatDurationSpy
+            .getMockImplementation()(resolvedIncidentMock.start_time, resolvedIncidentMock.end_time);
+        // @ts-ignore
+        expect(within(durationItem).getByText(durationValue)).toBeInTheDocument();
+    });
+
+    it("should display duration, but not end time (open incident)", () => {
+        render(
+            <IncidentDetails
+                incident={openIncidentMock}
+                onIncidentChange={onIncidentChange}
+            />
+        )
+
+        // Primary details section is rendered correctly
+        const primaryDetailsContainer = screen.getByTestId(/primary-details-container/i)
+        expect(primaryDetailsContainer).toBeInTheDocument()
+        expect(within(primaryDetailsContainer).getByText(/primary details/i)).toBeInTheDocument() // title
+        expect(within(primaryDetailsContainer).getByRole('list')).toBeInTheDocument()
+        expect(within(primaryDetailsContainer).getAllByRole('listitem').length).toBeGreaterThanOrEqual(6)
+
+        // End time item is not displayed
+        const endTimeItem = within(primaryDetailsContainer).queryByTitle(/end-time/i)
+        expect(endTimeItem).toBeNull()
+        expect(endTimeItem).not.toBeInTheDocument()
+
+        // Duration item is displayed
+        const durationItem = within(primaryDetailsContainer).getByTitle(/duration/i)
+        expect(durationItem).toBeInTheDocument()
+
+
+        // Duration value is displayed correctly
+        const durationValue = formatDurationSpy
+            .getMockImplementation()(openIncidentMock.start_time, openIncidentMock.end_time);
+        // @ts-ignore
+        expect(within(durationItem).getByText(durationValue)).toBeInTheDocument();
+    });
+
+    it("should display neither duration, nor end time (stateless incident)", () => {
+        render(
+            <IncidentDetails
+                incident={statelessIncidentMock}
+                onIncidentChange={onIncidentChange}
+            />
+        )
+
+        // Primary details section is rendered correctly
+        const primaryDetailsContainer = screen.getByTestId(/primary-details-container/i)
+        expect(primaryDetailsContainer).toBeInTheDocument()
+        expect(within(primaryDetailsContainer).getByText(/primary details/i)).toBeInTheDocument() // title
+        expect(within(primaryDetailsContainer).getByRole('list')).toBeInTheDocument()
+        expect(within(primaryDetailsContainer).getAllByRole('listitem').length).toBeGreaterThanOrEqual(6)
+
+        // End time item is not displayed
+        const endTimeItem = within(primaryDetailsContainer).queryByTitle(/end-time/i)
+        expect(endTimeItem).toBeNull()
+        expect(endTimeItem).not.toBeInTheDocument()
+
+        // Duration item is not displayed
+        const durationItem = within(primaryDetailsContainer).queryByTitle(/duration/i)
+        expect(durationItem).toBeNull()
+        expect(durationItem).not.toBeInTheDocument()
+    });
+});
+
```

Comment:
And I suppose if we're going to be super nitpicky remove the extra newline and only have 1 :P

Reply:
That one was litt of an oopsie 😂